task/amazon/resources/resource_security_group.go

```diff
@@ -0,0 +1,206 @@
+package resources
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"terraform-provider-iterative/task/amazon/client"
+	"terraform-provider-iterative/task/universal"
+)
+
+func NewSecurityGroup(client *client.Client, identifier string, defaultVPC *DefaultVPC, firewall universal.Firewall) *SecurityGroup {
+	s := new(SecurityGroup)
+	s.Client = client
+	s.Identifier = universal.NormalizeIdentifier(identifier, true)
+	s.Attributes = firewall
+	s.Dependencies.DefaultVPC = defaultVPC
+	return s
+}
+
+type SecurityGroup struct {
+	Client       *client.Client
+	Identifier   string
+	Attributes   universal.Firewall
+	Dependencies struct {
+		*DefaultVPC
+	}
+	Resource *types.SecurityGroup
+}
+
+func (s *SecurityGroup) Create(ctx context.Context) error {
+	switch err := s.Read(ctx); {
+	case err == nil:
+		return nil
+	case err != universal.NotFoundError:
+		return err
+	}
```

Comment:
```suggestion
func (s *SecurityGroup) Create(ctx context.Context) error {
	if err := s.Read(ctx); err != universal.NotFoundError {
		return err
	}
```

Reply:
you like `switch` too much... I have been always more a playstation fan 😁 
Seriously... `switch` is smelly, inelegant and should be removed from the mankind. Period